schema/deploy/shipping/views.sql

```diff
@@ -2911,4 +2913,200 @@ comment on view shipping.uw_reopening_ehs_reporting_v1 is
+                when tier = '1' then 'tier_1_surveillance'
+                when tier in ('2', '3') then 'tier_2_and_3_surveillance'
+                else null
+            end as priority_reason
+        from uw_enrollments
+        join uw_individuals using (individual)
+        -- Filter to enrollments that have never been invited to test or last invite was over 7 days before today
+        where (latest_invite_date is null or latest_invite_date < current_date - interval '7 days')
+        -- Filter to enrollments have never had a sample collected or last sample collection was over 7 days before today
+        and (latest_collection_date is null or latest_collection_date < current_date - interval '7 days')
+    ),
+
+    /**
+    Select encounters for surge testing purposes.
+    Handled separately from daily attestation encounters because we want to
+    include all surge selected participants regardless of previous testing.
+    **/
+    surge_testing as (
+        select
+            uw_encounters.redcap_project_id,
+            uw_encounters.redcap_record_id,
+            uw_encounters.redcap_event_name,
+            uw_encounters.redcap_repeat_instance,
+            uw_encounters.encountered,
+            individual,
+            tier,
+            latest_invite_date,
+            latest_collection_date,
+            6 as priority,
+            surge_selected as priority_reason
+        from uw_encounters
+        join uw_individuals using (individual)
+        join uw_enrollments using (individual)
+        -- Filter for instances that have a surge_selected reason
+        where (surge_selected is not null and length(surge_selected) > 0)
+        -- Filter for instances that do no already have testing_trigger filled (null or empty string)
+        and (testing_trigger is null or testing_trigger = '')
+    )
+
+    select * from positive_daily_attestations
+    union all
+    select * from baseline_and_surveillance
+    union all
+    select * from surge_testing
+)
+;
+
+comment on view shipping.__uw_priority_queue_v1 is
+  'Identify all encounter instances which indicate need for testing by UW reopening study (contains duplicate individuals)';
+
+
+create or replace view shipping.uw_priority_queue_v1 as (
+    with distinct_individuals as (
+        select distinct on (individual)
+            redcap_project_id,
+            redcap_record_id,
+            redcap_event_name,
+            redcap_repeat_instance,
+            encountered,
+            individual,
+            tier,
+            latest_invite_date,
+            latest_collection_date,
+            priority,
+            priority_reason
+        from shipping.__uw_priority_queue_v1
+        -- Use the highest priority encounter that is the latest instance
+        order by individual, priority asc nulls last, redcap_repeat_instance desc
+    )
+
+    select *
+    from distinct_individuals
+    order by priority asc
```

Comment:
We should be explicit about how we want to break priority ties by specifying fallback columns to order by. Options I can think of (none mutually exclusive):

- Newest or oldest first by encountered (which one is better?)
- `age(latest_collection_date) desc nulls first` — Time since last collection (longest first)
- `age(latest_invite_date) desc nulls first` — Time since last invite (longest first)
- Randomly!
- Record id

Reply:
I thought I was doing the fallback to the latest instance with the CTE above this:
```sql
order by individual, priority asc nulls last, redcap_repeat_instance desc
```

Reply:
Ah, that's within an individual's reasons to get tested. I'm talking about across individuals within a single priority level.

Reply:
Ah, I see. I thought that would be decided by the id3c command, but I can see it being easier to manage if just defined within the view itself.

I think the first fallback is tier (tier 1 get higher priority), then last collection date and last invite date?

Reply:
> I thought that would be decided by the id3c command, but I can see it being easier to manage if just defined within the view itself.

Yeah, I think the view should provide the authoritative state of the queue. This will help keep the command simpler by separating concerns, as well as make it much simpler when manually inspecting or writing reports on the queue.

> I think the first fallback is tier (tier 1 get higher priority), then last collection date and last invite date?

Sounds reasonable to me, but we should confirm with Trevor. Confirming doesn't need to block initial implementation/deploy. We may also want a final arbiter that can't (or is very unlikely to) be a tie, like individual identifier. This is better than record id because it's both stable (unlike random num) and (mostly) evenly distributed (like random num).

Reply:
I've asked Trevor in #informatics.

Reply:
Updated with the fallback for priority in the order proposed on slack.